pkg/build/generator/generator.go

```diff
@@ -738,31 +735,17 @@ func (g *BuildGenerator) resolveImageSecret(ctx apirequest.Context, secrets []ka
 	if len(secrets) == 0 || imageRef == nil {
 		return nil
 	}
-	emptyKeyring := credentialprovider.BasicDockerKeyring{}
 	// Get the image pull spec from the image stream reference
 	imageSpec, err := g.resolveImageStreamDockerRepository(ctx, *imageRef, buildNamespace)
 	if err != nil {
 		glog.V(2).Infof("Unable to resolve the image name for %s/%s: %v", buildNamespace, imageRef, err)
 		return nil
 	}
-	for _, secret := range secrets {
-		secretsv1 := make([]kapiv1.Secret, 1)
-		err := kapiv1.Convert_api_Secret_To_v1_Secret(&secret, &secretsv1[0], nil)
-		if err != nil {
-			glog.V(2).Infof("Unable to make the Docker keyring for %s/%s secret: %v", secret.Name, secret.Namespace, err)
-			continue
-		}
-		keyring, err := credentialprovider.MakeDockerKeyring(secretsv1, &emptyKeyring)
-		if err != nil {
-			glog.V(2).Infof("Unable to make the Docker keyring for %s/%s secret: %v", secret.Name, secret.Namespace, err)
-			continue
-		}
-		if _, found := keyring.Lookup(imageSpec); found {
-			return &kapi.LocalObjectReference{Name: secret.Name}
-		}
+	s := buildutil.FindDockerSecret(secrets, imageSpec)
```

Comment:
may make more sense to just comment FindDockerSecret better.  (which in this case means "at all" since it's currently not commented).  i think i'll do that.